x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.gen.ts

```diff
@@ -205,3 +321,78 @@ export const RelatedIntegration = z.object({
 
 export type RelatedIntegrationArray = z.infer<typeof RelatedIntegrationArray>;
 export const RelatedIntegrationArray = z.array(RelatedIntegration);
+
+export type InvestigationFields = z.infer<typeof InvestigationFields>;
+export const InvestigationFields = z.object({
+  field_names: z.array(z.string()).min(1),
+});
+
+/**
+ * Defines the interval on which a rule's actions are executed.
+ */
+export type RuleActionThrottle = z.infer<typeof RuleActionThrottle>;
+export const RuleActionThrottle = z.union([
+  z.enum(['no_actions', 'rule']),
+  z.string().regex(/^[1-9]\d*[smhd]$/),
+]);
+
+/**
+ * The action frequency defines when the action runs (for example, only on rule execution or at specific time intervals).
+ */
+export type RuleActionFrequency = z.infer<typeof RuleActionFrequency>;
+export const RuleActionFrequency = z.object({
+  summary: z.boolean(),
+  notifyWhen: z.enum(['onActionGroupChange', 'onActiveAlert', 'onThrottleInterval']),
+  throttle: RuleActionThrottle.nullable(),
+});
+
+export type RuleAction = z.infer<typeof RuleAction>;
+export const RuleAction = z.object({
+  /**
+   * The action type used for sending notifications.
+   */
+  action_type_id: z.string(),
+  /**
+   * Optionally groups actions by use cases. Use `default` for alert notifications.
+   */
+  group: z.string(),
+  /**
+   * The connector ID.
+   */
+  id: z.string(),
+  /**
+   * Object containing the allowed connector fields, which varies according to the connector type.
+   */
+  params: z.object({}).catchall(z.unknown()),
+  uuid: z.string().optional(),
+  alerts_filter: z.object({}).catchall(z.unknown()).optional(),
```

Comment:
This is actually defined in the original schema as 

```ts
export const RuleActionAlertsFilter = t.partial({
  query: t.union([
    t.undefined,
    t.intersection([
      t.strict({
        kql: t.string,
        filters: t.array(
          t.intersection([
            t.type({
              meta: t.partial({
                alias: t.union([t.string, t.null]),
                disabled: t.boolean,
                negate: t.boolean,
                controlledBy: t.string,
                group: t.string,
                index: t.string,
                isMultiIndex: t.boolean,
                type: t.string,
                key: t.string,
                params: t.any,
                value: t.string,
              }),
            }),
            t.partial({
              $state: t.type({ store: t.any }),
              query: t.record(t.string, t.any),
            }),
          ])
        ),
      }),
      t.partial({ dsl: t.string }),
    ]),
  ]),
  timeframe: t.union([
    t.undefined,
    t.strict({
      timezone: t.string,
      days: t.array(
        t.union([
          t.literal(1),
          t.literal(2),
          t.literal(3),
          t.literal(4),
          t.literal(5),
          t.literal(6),
          t.literal(7),
        ])
      ),
      hours: t.strict({
        start: t.string,
        end: t.string,
      }),
    }),
  ]),
});
```

Reply:
On the alerting framework side, this field extends `SavedObjectAttributes` which essentially turns this type into `Record<string, any>`:

```ts
export interface AlertsFilter extends SavedObjectAttributes {
  query?: {
    kql: string;
    filters: Filter[];
    dsl?: string; // This fields is generated in the code by using "kql", therefore it's not optional but defined as optional to avoid modifying a lot of files in different plugins
  };
  timeframe?: AlertsFilterTimeframe;
}
```

So a more correct representation on our side will be `Record<string, unknown>`. Let's discuss if we want to make this type more strict on our side.

Reply:
Hey @XavierM, FYI @banderror and I discussed the `AlertsFilter` type yesterday. We agreed that we don't want to expose this type through our API, which means we won't document it in OpenAPI. The plan is to keep it as `Record<string, unknown>` on our end. This gives us the flexibility to develop a better public interface for this functionality down the line. I think we also need to discuss which parts of the Alerting framework can be documented and exposed from the Security Solution API so we do not end up with documenting stuff that is under active development and subject to breaking changes.